x/fees/keeper/fee_decorator_test.go

```diff
@@ -67,18 +95,23 @@ func (suite *KeeperTestSuite) TestFeeDecorator() {
 				nil,
 				"",
 			),
-			shouldErr: false,
+			shouldErr:  false,
+			zeroHeight: false,
+			simulate:   true,
 		},
 	}
 
 	for _, tc := range testsCases {
 		ctx, _ := suite.ctx.CacheContext()
 		if tc.store != nil {
+			if tc.zeroHeight {
+				ctx = suite.ctx.WithBlockHeight(0)
+			}
```

Comment:
This is not how we should handle things here, as it makes things complicated. Instead what we should do is have a new optional field inside the `testCases` structure: 
```go
malleate func(ctx sdk.Context) sdk.Context
```

And then use it as follows:
```go
ctx, _ := suite.ctx.CacheContext()
if tc.malleate != nil {
  ctx = tc.malleate(ctx)
}
if tc.store != nil {
  tc.store(ctx)
}
```